content/docs/user-guide/checkpoints.md

```diff
@@ -0,0 +1,423 @@
+        summary: true
+        html: true
+```
+
+The checkpoints need to be enabled in DVC at the pipeline level. The
+`-c / --checkpoint` option of the `dvc stage add` command defines the checkpoint
+file or directory. This will create a new stage with checkpoints enabled, but
+does not update existing stages.
+
+The rest of the command `dvc stage add` sets up our dependencies for running the
+training code, which parameters we want to track (which are defined in the
+_params.yaml_), some configurations for our plots and showing the training
+metrics.
+
+We'll manually add a slight modification to the way our plots are handled. Add
+the following lines to your _dvc.yaml_. This just changes the way our results
+are shown after we've run some experiments.
+
+```yaml
+plots:
+  - predictions.json:
+      cache: false
+      template: confusion
+      x: actual
+      y: predicted
+```
+
+Now that you know how to enable checkpoints in a DVC pipeline, let's move on to
+setting up checkpoints in your code.
+
+## Registering checkpoints in your code
+
+Take a look at the _train.py_ file and you'll see how we train a convolutional
+neural network to classify handwritten digits. The main area of this code most
+relevant to checkpoints is when we iterate over the training epochs.
+
+This is where DVC will be able to track our metrics over time and where we will
+add our checkpoints to give us the points in time with our model that we can
+switch between.
+
+Now we need to enable checkpoints at the code level. We are interested in
+tracking the metrics along with each checkpoint, so we'll need to add a few
+lines of code.
+
+In the `train.py` file, import the `dvclive` package with the other imports:
+
+```bash
+import dvclive
+```
+
+Then update the following lines of code in the `main` method inside of the
+training epoch loop.
+
+```python
+for k, v in metrics.items():
+    print('Epoch %s: %s=%s'%(i, k, v))
+    dvclive.log(k, v)
+dvclive.next_step()
+```
+
+By adding the two `dvclive` methods to our training epoch loop, we are enabling
+checkpoints in our code and recording the training metrics at each of those
+checkpoints.
+
+## Running experiments
+
+With checkpoints enabled and working in our code, let's run the experiment
+again. You can run an experiment with the following command:
+
+```bash
+dvc exp run
+```
+
+You'll see output similar to this in your terminal while the training process is
+going on.
+
+```dvc
+Checkpoint experiment iteration 'd50c724'.
+Updating lock file 'dvc.lock'
+Checkpoint experiment iteration '29491a9'.
+Updating lock file 'dvc.lock'
+Checkpoint experiment iteration 'b3de55f'.
+Updating lock file 'dvc.lock'
+Checkpoint experiment iteration 'c4a46af'.
+Updating lock file 'dvc.lock'
+Checkpoint experiment iteration 'daf204c'.
+Updating lock file 'dvc.lock'
+Checkpoint experiment iteration 'bdb975c'.
+Updating lock file 'dvc.lock'
+Checkpoint experiment iteration '77dc46c'.
+Updating lock file 'dvc.lock'
+```
```

Comment:
I have a slightly different output format. Users can be confused by the difference.
```bash
.....
Epoch 2: loss=1.25374174118042
Epoch 2: acc=0.7738
Updating lock file 'dvc.lock'
Checkpoint experiment iteration '4505b31'.

file:///Users/dmitry/src/tmp/checkpoints-tutorial/dvclive.html
Epoch 3: loss=0.7242147922515869
Epoch 3: acc=0.8284
Updating lock file 'dvc.lock'
Checkpoint experiment iteration '013ea5b'.

file:///Users/dmitry/src/tmp/checkpoints-tutorial/dvclive.html
Epoch 4: loss=0.5083536505699158
Epoch 4: acc=0.8538
Updating lock file 'dvc.lock'
Checkpoint experiment iteration '7a62f2a'.
```